libs/components/core/src/lib/modules/dock/dock.service.ts

```diff
@@ -15,7 +15,7 @@ import { sortByStackOrder } from './sort-by-stack-order';
  * This service docks components to specific areas on the page.
  */
 @Injectable({
-  providedIn: 'root',
+  providedIn: 'any',
```

Comment:
This is the same issue that we have had to solve for Modals and others where the "providedIn: `root`" causes internationalization problems with dynamically generated components. This is the same fix that we did for modals in order to ensure that all the resource labels are output correctly.

Reply:
In this case, though, couldn't we be in a situation where we have multiple dock host elements? One (at least) for the root injector, and another dock host for each lazy-loaded module that gets spun up. Which component/service is affected by the dock service using `providedIn: 'root'`?

It might be worth looking into using `static` properties for the `dockRef` and `items` instead of `providedIn: 'any'`, but I'm unsure of all the repercussions of that change.